src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1053,6 +1152,67 @@ double VideoDecoder::getPtsSecondsForFrame(
   return ptsToSeconds(stream.allFrames[frameIndex].pts, stream.timeBase);
 }
 
+void VideoDecoder::convertFrameToBufferUsingSwsScale(
+    RawDecodedOutput& rawOutput) {
+  AVFrame* frame = rawOutput.frame.get();
+  int streamIndex = rawOutput.streamIndex;
+  enum AVPixelFormat frameFormat =
+      static_cast<enum AVPixelFormat>(frame->format);
+  StreamInfo& activeStream = streams_[streamIndex];
+  int outputWidth = activeStream.options.width.value_or(frame->width);
+  int outputHeight = activeStream.options.height.value_or(frame->height);
+  if (activeStream.swsContext.get() == nullptr) {
+    SwsContext* swsContext = sws_getContext(
+        frame->width,
+        frame->height,
+        frameFormat,
+        outputWidth,
+        outputHeight,
+        AV_PIX_FMT_RGB24,
+        SWS_BILINEAR,
+        nullptr,
+        nullptr,
+        nullptr);
+    const int *invTable, *table;
+    int srcRange, dstRange, brightness, contrast, saturation;
+    sws_getColorspaceDetails(
+        swsContext,
+        (int**)&invTable,
```

Comment:
This cast is only necessary because we declared `invTable` as a const pointer, but I don't see any value in doing so. We should declare it and `table` as non-const pointers, eliminating the need for the casts.